cli/cmd/main.go

```diff
@@ -587,3 +695,89 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+func DataCommand(c *cli.Context) error {
+	if c.String(dataFlagType) != "binary" && c.String(dataFlagType) != "tabular" {
+		return errors.Errorf("type must be binary or tabular, got %s", c.String("type"))
+	}
+
+	filter := &datapb.Filter{}
+	if c.StringSlice(dataFlagOrgs) != nil {
+		filter.OrgIds = c.StringSlice(dataFlagOrgs)
+	}
+	if c.String(dataFlagLocation) != "" {
+		filter.LocationId = c.String(dataFlagLocation)
+	}
+	if c.String(dataFlagRobotID) != "" {
+		filter.RobotId = c.String(dataFlagRobotID)
+	}
+	if c.String(dataFlagPartID) != "" {
+		filter.PartId = c.String(dataFlagPartID)
+	}
+	if c.String(dataFlagRobotName) != "" {
+		filter.RobotName = c.String(dataFlagRobotName)
+	}
+	if c.String(dataFlagPartName) != "" {
+		filter.PartName = c.String(dataFlagPartName)
+	}
+	if c.String(dataFlagComponentType) != "" {
+		filter.ComponentType = c.String(dataFlagComponentType)
+	}
+	if c.String(dataFlagComponentModel) != "" {
+		filter.ComponentModel = c.String(dataFlagComponentModel)
+	}
+	if c.String(dataFlagComponentName) != "" {
+		filter.ComponentName = c.String(dataFlagComponentName)
+	}
+	if c.String(dataFlagMethod) != "" {
+		filter.Method = c.String(dataFlagMethod)
+	}
+	if len(c.StringSlice(dataFlagMimeTypes)) != 0 {
+		filter.MimeType = c.StringSlice(dataFlagMimeTypes)
+	}
+
+	var start *timestamppb.Timestamp
+	var end *timestamppb.Timestamp
+	timeLayout := "2000-01-01T00:00:00.000"
```

Comment:
Do you need to specify the timezone here? I think it will assume that the provided time is UTC if unspecified in the format but not as familiar with the quirks of the time library

Reply:
I think you can also use `time.RFC3339` in place of the timeLayout as specified in [here](https://pkg.go.dev/time#Parse) - should allow for timezones

Reply:
That's much cleaner, changed. thanks!